### ISSUE TITLE
Publishing fails to release Sonatype repository

### ISSUE THREAD

**Issue by kubukoz** [VirtusLab/scala-cli#1682]
**Version(s)**
0.1.18

**Describe the bug**

Seeing this when trying to publish a library:

```
Exception in thread "main" java.lang.Exception: java.lang.Exception: Failed to get https://s01.oss.sonatype.org/service/local/staging/profiles/44c2ce0e190826/promote (http status: 500, response: {"errors":[{"id":"*","msg":"Unhandled: Staging repository is already transitioning: orgpolyvariant-1045"}]})
```

Seeing the failure I checked Nexus and it claimed the repo is closed, so it just needed to be released. I did that manually in the Nexus UI and it [seems to have worked](https://s01.oss.sonatype.org/content/repositories/releases/org/polyvariant/smithy4s-utils_2.13/0.0.1/).

**To Reproduce**

Given the nature of publishing it'd be kind of hard to reproduce reliably and independently, but I used [this GitHub workflow](https://github.com/polyvariant/smithy4s-utils/blob/5955d7a56f0849ab27e175301e3edb34827f16e4/.github/workflows/publish.yml). The repository at that state also contains the scala-cli config in use.

**Expected behaviour**
Publishing happens successfully

**Comment by kubukoz** [comment]
Also, if there's a simpler way to achieve what I'm doing (getting the gpg secret from a base64-encoded environment variable), I'd appreciate any hints on such simplifications :)

**Comment by kubukoz** [comment]
Looks like it worked the second time (hell, it worked for [3 Scala versions in a row](https://github.com/polyvariant/smithy4s-utils/actions/runs/3690430208/jobs/6247414638)!), so that must've been some sort of race condition / temporary quirk. Probably worth looking into, nevertheless

**Comment by MartinHH** [comment]
This just happened to me as well (Scala CLI version: 1.0.4) when I tried `scala-cli publish . --native` (from my local machine) with [this configuration](https://github.com/MartinHH/scalacheck-derived/blob/v0.4.2/project.scala). I could reproduce this twice (dropped the staging repository and retried), then released manually in the UI ([which seems to have worked](https://s01.oss.sonatype.org/content/repositories/public/io/github/martinhh/scalacheck-derived_native0.4_3/0.4.2/)).

It might be worth noting that between the first failure and the failing retries, I successfully executed `scala-cli publish . --js` (on first try), so I guess it's not caused by a totally random race condition.
